extensions/xcode/src/open-url-in-simulator.command.tsx

```diff
@@ -1,10 +1,53 @@
 import { XcodeSimulatorService } from "./services/xcode-simulator.service";
-import { Clipboard, LaunchProps } from "@raycast/api";
+import { closeMainWindow, launchCommand, LaunchProps, LaunchType, showHUD, showToast, Toast } from "@raycast/api";
+import {
+  XcodeSimulatorOpenUrlError,
+  XcodeSimulatorOpenUrlErrorReason,
+} from "./models/xcode-simulator/xcode-simulator-open-url-error.model";
 
-interface OpenSimulatorArguments {
-  url?: string;
-  udid?: string;
-}
-
-export default async (props: LaunchProps<{ arguments: OpenSimulatorArguments }>) =>
-  XcodeSimulatorService.openUrl(props.arguments.url ?? (await Clipboard.readText()), undefined);
+export default async (props: LaunchProps<{ arguments: { url: string } }>) => {
+  // Show loading toast
+  const toast = await showToast({
+    style: Toast.Style.Animated,
+    title: "Opening URL in Simulator",
+  });
+  try {
+    // Try to open url
+    await XcodeSimulatorService.openUrl(props.arguments.url);
+    // Close main window
+    await closeMainWindow();
+  } catch (error) {
+    // Check if error is an instance of XcodeSimulatorOpenUrlError
+    if (error instanceof XcodeSimulatorOpenUrlError) {
+      // Switch on reason
+      switch (error.reason) {
+        case XcodeSimulatorOpenUrlErrorReason.badUrl:
+          // Show failure toast
+          toast.style = Toast.Style.Failure;
+          toast.title = "Please enter a valid url";
+          break;
+        case XcodeSimulatorOpenUrlErrorReason.bootedSimulatorMissing:
+          // Hide toast
+          await toast.hide();
+          // Show info hud
+          await showHUD("Please boot a simulator to open a url");
+          // Launch manage simulators command in order to boot a simulator
+          await launchCommand({ name: "manage-simulators.command", type: LaunchType.UserInitiated });
```

Comment:
I find it to be possibly weird here. By doing so, on UX perspective we could expect that the url opens right after the selected simulator finished booting. That's why I prefered a classic error message. WDYT?

Reply:
I have refactored the implementation to a failure toast with a primary action allowing to easily open the "Manage Simulators" command 👌 (https://github.com/qeude/extensions/pull/1/commits/33909b321812a402bbf4800bbae7c62742f75db2)

<img width="478" alt="Bildschirm­foto 2023-02-05 um 16 16 40" src="https://user-images.githubusercontent.com/11733014/216828110-bc8f33bc-360b-4300-99df-da6e1e66c140.png">